index.js

```diff
@@ -6,16 +6,31 @@ const QuickLru = require('quick-lru');
 const has = (array, key) => array.some(x => typeof x === 'string' ? x === key : x.test(key));
 const cache = new QuickLru({maxSize: 100000});
 
+const isObject = value =>
+	typeof value === 'object' &&
+	value !== null &&
+	!(value instanceof RegExp) &&
+	!(value instanceof Error) &&
+	!(value instanceof Date);
```

Comment:
`Math` is an object too. Could use this module https://github.com/sindresorhus/is-plain-obj if you want to cover everything.

Reply:
This only reproduces behavior from `map-obj`.
Also, don't see what's the problem if `Math` passes this test.

Reply:
It was added in https://github.com/sindresorhus/map-obj/commit/df69cde79f9d8cf08a633fc6858cc637c1a0f776, I don't think it's that optimized and I assume @sindresorhus would use the other method today.

> Though I don't really know why map-obj excludes RegExp, Error and Date, but the comment says it's fine tuned for this case.

"This case" probably means checking excluding all objects except "plain" ones and arrays.

> Also, don't see what's the problem if Math passes this test.

It's not meant to since you can't map over it. We only want objects and arrays.

Reply:
For the sake of correctness, you can map over `Math`, though there will be 0 keys unless you modify it. 

As for the implementation in `map-obj`, I won't risk making any assumptions regarding reasons it behaves this way right now, but regardless of that both thing should be using the same check in order for result to be consistent.

If `is-plain-object` is better, it's still a breaking change for `map-obj`, so probably should be discussed in `map-obj` project(with a follow up patch here in case of changes).

Reply:
Yeah, I might be understanding it wrong as well. I'd keep it as is for now.